docs/rfc/rfc-6.adoc

```diff
@@ -0,0 +1,228 @@
+sense to mark it as inactive and ineligible for rewards one more time.
+
+IMPORTANT: There should be no token incentives for calling this function by
+the group. The only incentive should be their goodwill to save some rewards by
+DAO. Moreover, in the case of off-chain heartbeat failure, clients are trying to
+avoid slashing in the future. Marking operators as ineligible for rewards should
+not increase the rewards of other group members! Rewards saved on inactive
+operators marked as ineligible should return back to the DAO. The DAO should do
+something else with those extra tokens other than increase rewards. We do not
+want to incentivize voting for your own salary.
+
+[[maintainer-incentives]]
+==== Maintainer incentives
+
+For the first release, the role of maintainers will belong to the willing
+off-chain clients approved by DAO or council. Every client will have maintainer
+functionality built-in and enabled in the configuration file.
+
+We will implement a simple reimbursement contract for maintainers proxying calls
+to Bridge functions. Only approved maintainers should be able to use the proxy
+contract. Maintainers are expected to follow round-robin distribution of work
+but the order is not going to be enforced on-chain. DAO/council will have the
+functionality of removing maintainers not performing their duties.
+DAO/council may decide to distribute additional T rewards for maintainers. This
+functionality should be supported by the Maintainer contract.
+
+Eventually, the role of maintainers can be delegated to another network such as
+Gelato.
+
+
+```
+┌────────────────────────────────────────┐   calls  ┌─────────────────────────┐
+|               Maintainer               |─────────▶|          Bridge         |
+└────────────────────────────────────────┘          └─────────────────────────┘  
+| submitSweeProof() onlyMaintainer       |          | submitSweepProof()      | 
+| submitRedemptionProof() onlyMaintainer |          | submitRedemptionProof() | 
+| startDkg() onlyMaintainer              |          | startDkg()              | 
+└────────────────────────────────────────┘          └─────────────────────────┘
+```
+
+===== Submit sweep proof
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+===== Submit redemption proof
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+===== Start DKG
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers. 
+
+===== Report DKG timeout
+
+ETH reimbursed, no additional reward to disincentivize front-running between
+maintainers.
+
+==== Third party incentives
+
+
+All transactions in this section must be supported by the off-chain client. We
+are not concerned about front-running. Front-running may happen and it is even
+desirable to some extent because these transactions protect the health of the
+network.
+
```

Comment:
Could we also list 'Notify relay timeout' here?

Reply:
This RFC is for tBTC v2 and relay entry timeout is not a part of it. I'd rather not mix them there.